src/reflect/value.go

```diff
@@ -689,6 +689,27 @@ func (v Value) Cap() int {
 	}
 }
 
+//go:linkname mapclear runtime.hashmapClear
+func mapclear(p unsafe.Pointer)
+
+// Clear clears the contents of a map or zeros the contents of a slice
+//
+// It panics if v's Kind is not Map or Slice.
+func (v Value) Clear() {
+	switch v.typecode.Kind() {
+	case Map:
+		mapclear(v.pointer())
+	case Slice:
+		hdr := (*sliceHeader)(v.value)
+		elemSize := v.typecode.underlying().elem().Size()
+		for i := uintptr(0); i < hdr.len; i++ {
```

Comment:
Shouldn't this just be a single `memzero` call instead of one call for each element in the slice?

Reply:
Updated.